### PR TITLE
rest: Improve tests and documention of /headers and /block

### PR DESCRIPTION
**Pull request by promag**
Follow up of #15107.

### CODE REVIEW

**Comment by laanwj** [comment]
utACK

**Comment by promag** [comment]
Now with /rest/block documentation and test improvement.

**Comment by hebasto** [comment]
utACK 7cf994d5cfd53dcff76ebd0e0007e3477a7570e8
nit: I'm inclined do not use contractions in the docs.

File: test/functional/interface_rest.py [hunk 1]
```diff
@@ -201,6 +201,16 @@ def run_test(self):
...
         bb_hash = self.nodes[0].getbestblockhash()
 
+        # Check result if block does not exists
+        assert_equal(self.test_rest_request('/headers/1/0000000000000000000000000000000000000000000000000000000000000000'), [])
+        self.test_rest_request('/block/0000000000000000000000000000000000000000000000000000000000000000', status=404, ret_type=RetType.OBJ)
+
+        # Check result if block is not in the active chain
+        self.nodes[0].invalidateblock(bb_hash)
+        assert_equal(self.test_rest_request('/headers/1/{}'.format(bb_hash)), [])
+        self.test_rest_request('/block/{}'.format(bb_hash))
```
Comment (Empact) [inline test/functional/interface_rest.py]: nit: could be more clear to make explicit what default args you're finding most relevant here

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15177/commits/7cf994d5cfd53dcff76ebd0e0007e3477a7570e8 could squash

File: test/functional/interface_rest.py (hunk 1 above)
Comment (promag) [reply test/functional/interface_rest.py]: Maybe explicit `status=200`?

**Comment by laanwj** [comment]
utACK 7cf994d5cfd53dcff76ebd0e0007e3477a7570e8

**Pull request merged**